### PR TITLE
39 blogposts logan

### PR DESCRIPTION
**Pull request by kiserd**
I want to note that I acknowledge the UI looks ROUGH. I'm not used to operating outside a full stack capacity, but I'll let the front-end folks polish things as they see fit (@jmm28).
image

Should provide simple CRUD functionality without any sort of auth (pending `flask-principal` branch).

`C`
image

`R`
image

`U`
image
image

`D`
image

Once again, things are looking janky. That said, I don't want to step on styling toes :P

Finally, I can help out with merge conflicts. I messed up pushing the `.pyc` files early. `blogPosts.py` should allow the incoming changes to persist.

### CODE REVIEW

File: app/templates/profile.html [hunk 1]
```diff
@@ -1,4 +1,4 @@
-{% extends "base.html" %}
+{% extends "layout.html" %}
```
Comment (kiserd) [inline app/templates/profile.html:1]: Looked like we just had a little typo in here. I probably should have left this for a different branch, but this was pretty early in my journey. Wanted to confirm that my sign up / login was working.

File: app/templates/blog/update.html [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        <form method="POST" action="/blog/{{ post.id }}/update">
+            <div class="mb-3">
+                <label for="title" class="form-label">Title</label>
+                <input
+                class="form-control"
+                type="text"
+                name="title"
+                id="title"
+                placeholder="{{ post.title }}"
+                value="{{ post.title }}">
```
Comment (kiserd) [inline app/templates/blog/update.html:15]: We don't need to leave things like this. I was getting dizzy scrolling to the right!

File: app/blogPosts.py [hunk 3]
```diff
@@ -1,52 +1,87 @@
...
 
-@blogPosts.route("/post/<int:post_id>/delete", methods=['POST'])
-@login_required
+@blogPosts.route("/blog/<int:post_id>/delete", methods=['GET'])
+# Logan Kiser: hang tight on requiring credentials for flask-principal branch
+# @login_required
 def delete_post(post_id):
-    post = Post.query.get_or_404(post_id)
-    if post.author != current_user:
-        abort(403)
```
Comment (kiserd) [inline app/blogPosts.py:48]: Plan to add similar checking in the flask-principal branch.

File: app/blogPosts.py [hunk 4]
```diff
@@ -1,52 +1,87 @@
...
-        form.title.data = post.title
-        form.content.data = post.content
-    return render_template('create_post.html', title='Update Post',
-                           form=form, legend='Update Post')
+        return redirect(url_for('blogPosts.display_post', post_id=post.id))
+    # populate form with existing data and allow user to alter blog post
+    return render_template('blog/update.html', post=post, links=get_links())
 
 
-@blogPosts.route("/post/<int:post_id>/delete", methods=['POST'])
```
Comment (kiserd) [inline app/blogPosts.py:43]: little hack to get things to work nicely with a simple `<a>` tag

File: app/__init__.py [hunk 5]
```diff
@@ -29,6 +39,10 @@ def load_user(user_id):
     from .auth import auth as auth_blueprint
     app.register_blueprint(auth_blueprint)
 
+    # blueprint for blogpost routes in app
+    from .blogPosts import blogPosts as blogPosts_blueprint
+    app.register_blueprint(blogPosts_blueprint)
```
Comment (kiserd) [inline app/__init__.py:44]: Tossed this in here per Stuart's comment on the last PR.

File: app/__init__.py [hunk 6]
```diff
@@ -10,15 +10,25 @@ def create_app():
...
     login_manager.init_app(app)
 
     from .models import User
 
+    ######
+    # logan kiser: troubleshooting db issues, will delete before submitting
+    #              pull request.
+    # with app.app_context():
+        # db.create_all()
+    ######
```
Comment (kiserd) [inline app/__init__.py:30]: I'll touch base with others to find a better way of doing this. Most likely pushing a context via command line.

File: app/blogPosts.py [hunk 7]
```diff
@@ -1,52 +1,87 @@
...
-    post = Post.query.get_or_404(post_id)
-    if post.author != current_user:
-        abort(403)
+    post = BlogPost.query.get_or_404(post_id)
     db.session.delete(post)
     db.session.commit()
     flash('Your post has been deleted!', 'success')
-    return redirect(url_for('home'))
\ No newline at end of file
+    return redirect(url_for('blogPosts.display_blog_home'))
```
[the remark this answers is not in the document]
Comment (stuart909) [reply app/blogPosts.py]: blog_home is better.  url_for kind of insinuates page loads, don't you think?

File: app/blogPosts.py (hunk 3 above)
Comment (stuart909) [reply app/blogPosts.py:48]: Yes.  I would prefer a group membership model.

File: app/blogPosts.py (hunk 4 above)
Comment (stuart909) [reply app/blogPosts.py:43]: I don't know how I feel about this one.  Such an obvious delete method being exposed to the world wide web begs for exploitation.

**Comment by stuart909** [comment]
Don't commit your pycache and use -f when you push/merge.  We will eventually stop pushing databases, encryption keys, and pycache.

File: app/blogPosts.py (hunk 4 above)
Comment (kiserd) [reply app/blogPosts.py:43]: Agreed! I didn't think this was live (yet). I didn't want to go too crazy tweaking the templates. I'll alter things in favor of best practices and @jmm28 can repackage to their heart's desire!

**Comment by stuart909** [comment]
This doesn't do a few things:

1. Sign in / Sign out is borked.  The sign out doesn't load.
2. The posts mechanism doesn't load when you login.
3. There is no posts link when I am logged in.

Let me double check your permissions on Tobuwebflask.  There is a working model you were supposed to copy the example from my corporate version.

**Comment by stuart909** [comment]
Don't merge this.

**Review by stuart909 — CHANGES_REQUESTED** [review]
You are making good progress.  Though it seems some of the site functionality got broken in your update.  See previous comments.

File: app/blogPosts.py (hunk 7 above)
[the remark this answers is not in the document]
Comment (kiserd) [reply app/blogPosts.py]: This is my inexperience shining bright. Kabir used `url_for` and I figured he had good reason for it. I can switch things up.

**Comment by kiserd** [comment]
On it! I haven't seen some of the comments until moments ago.

**Comment by kiserd** [comment]
> This doesn't do a few things:
> 
> 1. Sign in / Sign out is borked.  The sign out doesn't load.
> 2. The posts mechanism doesn't load when you login.
> 3. There is no posts link when I am logged in.
> 
> Let me double check your permissions on Tobuwebflask. There is a working model you were supposed to copy the example from my corporate version.

On it! I overlooked several of your comments earlier (not sure how).
(1) Not sure how I missed this. I was signing up and logging in/out a ton. I'll take a look.
(2) Odd. I'll take a look!
(3) Once again, I was trying not to step on front-end toes. I was manually entering the endpoint(s) into the browser.

**Comment by kiserd** [comment]
> This doesn't do a few things:
> 
> 1. Sign in / Sign out is borked.  The sign out doesn't load.
> 2. The posts mechanism doesn't load when you login.
> 3. There is no posts link when I am logged in.
> 
> Let me double check your permissions on Tobuwebflask. There is a working model you were supposed to copy the example from my corporate version.

Hey @stuart909,

I took another look through things. I definitely misunderstood the ask a bit

(1 & 2) I think the Sign in / Sign out was already borked on this branch. The nav bar doesn't react to user authentication in either `39-blogPosts` or `main`. I took a look at Jasmine's `37-profile-page` branch and I think she has worked lil' bit of this out over there. Regardless, I made the desired changes in this branch per the requested change.

Prior to logging in
image

After logging in
image

It is worth noting that our `Logout` route seems to be borked as well. However, I thought it would be more appropriate to fix that in another branch. That said, upon request, I'm happy to lump it in here with the blog post stuff :)

(3) Easy-peasy, nav bar has the link now.

**Review by stuart909 — APPROVED** [review]
I will approve these changes, but we have a ways to go before blogposts is considered incorporated.  There is currently no blogposts view being offered up by our routes

**Comment by kiserd** [comment]
> I will approve these changes, but we have a ways to go before blogposts is considered incorporated. There is currently no blogposts view being offered up by our routes

Shoot! I can take a closer look when I get up in the morning. I tried to model everything in the `tobuwebflask` image, but obviously am missing something. I'll either merge this into the original `39-blogPosts` branch and create/checkout another branch to resolve this OR stick the changes in this branch and re-request review.

I'm not sure what I'm missing, but I bet it will be super apparent after sleeping.

**Comment by kiserd** [comment]
> I will approve these changes, but we have a ways to go before blogposts is considered incorporated. There is currently no blogposts view being offered up by our routes

I am hoping that this has something to do with my having commented out the
`with app.app_context():
    db.create_all()`
in the `app/__init__.py` file. I'm not sure how folks have been building out the db schema. I know there are ways to do so via the command line. So long as I create the tables, I haven't had an issue with the views - we can touch base and take a look!

**Pull request merged**